### PR TITLE
tmate contributor training blog

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

Today's Tmate Contributor training blog

Review rendered at https://20241023-tmate.ddev-com-front-end.pages.dev/blog/tmate-github-actions-contributor-training/

@stasadev may add more info about using act.

### CODE REVIEW

**Comment by tyler36** [comment]
The video links to the same video as "ddev debug test" training session. 
Just a reminder to update the video link once youtube is update.

**Comment by rfay** [comment]
Thanks. I haven't uploaded or linked yet.

**Comment by rfay** [comment]
Now it's better @tyler36 , thanks for taking a look!

**Comment by tyler36** [comment]
Confirmed. Blog links now links to "Debugging GitHub Actions Workflows with Tmate" video.

File: src/content/blog/tmate-github-actions-contributor-training.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Why do we need Tmate?
+
+Often it's hard to understand what has happened with an test because all we see in GitHub's web UI is the output, and we can't interact with it. And trying to recreate the test environment is sometimes fine, but sometimes it's hard to recreate the test. GitHub runners have different memory configuration, disk space, and different packages installed, and they're typically running AMD64 Ubuntu, which may not be something we have easy access to.
+
+## Alternatives to Tmate
+
+1. We normally will try to understand a test failure by running it locally.
+2. Running in a similar Linux/AMD64 system like GitPod is a pretty easy option. 
+3. [nektos/act](https://github.com/nektos/act) is another recommended competitor to Tmate. It uses Docker and a Docker image to run an action on your local machine. I haven't had luck with it when I've tried it.
```
Comment (stasadev) [inline src/content/blog/tmate-github-actions-contributor-training.md:34]:
````suggestion
3. [nektos/act](https://github.com/nektos/act) is another recommended competitor to Tmate. It uses Docker and a Docker image to run an action on your local machine. I haven't had luck with it when I've tried it.  Stas's remarks:
    - act is useful to run simple jobs, for example, I want to test a GitHub API call using `actions/github-script`.
    - The most important thing is to choose [the right image](https://github.com/nektos/act/issues/2219) for each specific job to make it work.
    - I remove everything from `.github/workflows/test.yml`, leaving only code related to the job I want to test:

    ```yaml
    jobs:
      build:
        runs-on: ubuntu-latest
        steps:
          - name: Get my GitHub repositories
            uses: actions/github-script@v7
            with:
              github-token: ${{ secrets.GITHUB_TOKEN }}
              script: |
                const { data } = await github.rest.search.repos({q: 'user:stasadev'})
                console.log(data)
                const fs = require('fs');
                fs.writeFileSync('data.json', JSON.stringify(data, null, 2)); 
    ```

    And then I can run `act` in the project root:

    ```bash
    act -P ubuntu-latest=catthehacker/ubuntu:act-latest --bind --job build -s GITHUB_TOKEN=my_token
    ```

    It prints out the contents of the `data` in my terminal, and writes it to `data.json` in the current directory, which makes it much easier to debug.
````

**Review by stasadev — APPROVED** [review]
Looks good to me, I prepared a suggestion for `act`.
Feel free to change it.

**Review by tyler36 — APPROVED** [review]

**Comment by stasadev** [comment]
I've updated `act` section.

**Comment by rfay** [comment]
Thanks @stasadev , I may still add an image to this.

**Pull request merged**